pageserver/src/layered_repository.rs

```diff
@@ -598,16 +598,35 @@ impl Timeline for LayeredTimeline {
         trace!("list_rels called at {}", lsn);
 
         // List all rels in this timeline, and all its ancestors.
-        let mut all_rels = HashSet::new();
+        let mut all_rels_map: HashMap<RelTag, bool> = HashMap::new();
```

Comment:
This could use more comments. Note in comment that the 'bool' here indicates whether the relation was dropped or not.